scripts/irods/test/test_iput.py

```diff
@@ -886,6 +887,50 @@ def test_large_put(self):
         self.object_name = 'test_large_put'
         self.run_tests(self.object_name, 5120000)
 
+    def test_iput_with_updated_checksum_read_buffer_size_in_bytes__issue_7947(self):
+        self.object_name = 'test_iput_with_updated_checksum_read_buffer_size_in_bytes__issue_7947'
+        irodsctl = IrodsController()
+        server_config_filename = paths.server_config_path()
+
+        # load server_config.json to inject a new rule base
+        with open(server_config_filename) as f:
+            svr_cfg = json.load(f)
+
+        # update the checksum_read_buffer_size_in_bytes to 10 MiB
+        svr_cfg['advanced_settings']['checksum_read_buffer_size_in_bytes'] = 10*1024*1024
+
+        # dump to a string to repave the existing server_config.json
+        new_server_config = json.dumps(svr_cfg, sort_keys=True, indent=4, separators=(',', ': '))
+
+        # create a 100 MiB file
+        file_size = 100*1024*1024
+        local_file = os.path.join(self.admin.local_session_dir, self.object_name)
+        lib.make_file(local_file, file_size)
```

Comment:
File needs to be removed when the test ends (regardless of success or failure).

Reply:
Missed that.  Done.